src/app/meetups/meetups.component.ts

```diff
@@ -203,7 +205,17 @@ export class MeetupsComponent implements OnInit, AfterViewInit, OnDestroy {
 
   countSelectedShelf(selected: any) {
     const { inShelf, notInShelf } = this.userService.countInShelf(selected, 'meetupIds');
-    this.selectedJoined = inShelf;
     this.selectedNotJoined = notInShelf;
+    this.selectedJoined = inShelf;
+    let index, meetupDate;
+    for (let i = 0; i < selected.length; i++) {
+      index = this.meetups.data.findIndex(x => x._id === selected[i]);
+      meetupDate = this.meetups.data[index].endDate ? this.meetups.data[index].endDate : this.meetups.data[index].startDate;
+      if (this.dateNow > meetupDate) {
+        this.selectedNotJoined--;
+        selected.splice(i, 1);

```

Comment:
I think this is causing strange behavior.  I can't leave an old meetup (it says "You have left 0 meetups") and after a refresh I'm still a member of that meetup.

Also it's causing the select all button to flicker when select all is clicked